prow/kube/prowjob.go

```diff
@@ -48,11 +48,13 @@ const (
 )
 
 type ProwJob struct {
-	APIVersion string        `json:"apiVersion,omitempty"`
-	Kind       string        `json:"kind,omitempty"`
-	Metadata   ObjectMeta    `json:"metadata,omitempty"`
-	Spec       ProwJobSpec   `json:"spec,omitempty"`
-	Status     ProwJobStatus `json:"status,omitempty"`
+	APIVersion   string        `json:"apiVersion,omitempty"`
+	Kind         string        `json:"kind,omitempty"`
+	Metadata     ObjectMeta    `json:"metadata,omitempty"`
+	Spec         ProwJobSpec   `json:"spec,omitempty"`
+	Status       ProwJobStatus `json:"status,omitempty"`
+	ParentStatus ProwJobStatus `json:"parent_status,omitempty"`
+	ParentJob    string        `json:"parent_job,omitempty"`
```

Comment:
That's an option, but I wanted to avoid restricting it to build jobs so the flow is:
Job writes to logs bucket alongside started.json, finished.json, etc with data for the child job to consume (say `child_data.json` or something), prow injects the parent job name / buildID into the child when it starts and the child goes to logs/$PARENT_JOB_NAME/$PARENT_BUILD_ID/ and uses child_data.json to do whatever (in this get the build location).